app/main/views/login.py

```diff
@@ -0,0 +1,355 @@
+                    "error {error} email_hash {email_hash}",
+                    extra={'error': six.text_type(e),
+                           'email_hash': hash_email(user.email_address)})
+                abort(503, "Failed to send password reset")
+
+            current_app.logger.info(
+                "login.reset-email.sent: Sending password reset email for "
+                "supplier_id {supplier_id} email_hash {email_hash}",
+                extra={'supplier_id': user.supplier_id,
+                       'email_hash': hash_email(user.email_address)})
+        else:
+            current_app.logger.info(
+                "login.reset-email.invalid-email: "
+                "Password reset request for invalid supplier email {email_hash}",
+                extra={'email_hash': hash_email(email_address)})
+
+        flash('email_sent')
+        return redirect(url_for('.request_password_reset'))
+    else:
+        template_data = main.config['BASE_TEMPLATE_DATA']
+        return render_template("auth/request-password-reset.html",
+                               form=form,
+                               **template_data), 400
+
+
+@main.route('/reset-password/<token>', methods=["GET"])
+def reset_password(token):
+    decoded = decode_password_reset_token(token, data_api_client)
+    if not decoded:
+        return redirect(url_for('.request_password_reset'))
+
+    email_address = decoded["email"]
+
+    template_data = main.config['BASE_TEMPLATE_DATA']
+    return render_template("auth/reset-password.html",
+                           email_address=email_address,
+                           form=ChangePasswordForm(),
+                           token=token,
+                           **template_data), 200
+
+
+@main.route('/reset-password/<token>', methods=["POST"])
+def update_password(token):
+    form = ChangePasswordForm()
+    decoded = decode_password_reset_token(token, data_api_client)
+    if not decoded:
+        return redirect(url_for('.request_password_reset'))
+
+    user_id = decoded["user"]
+    email_address = decoded["email"]
+    password = form.password.data
+
+    if form.validate_on_submit():
+        if data_api_client.update_user_password(user_id, password, email_address):
+            current_app.logger.info(
+                "User {user_id} successfully changed their password",
+                extra={'user_id': user_id})
+            flash('password_updated')
+        else:
+            flash('password_not_updated', 'error')
+        return redirect(url_for('.render_login'))
+    else:
+        template_data = main.config['BASE_TEMPLATE_DATA']
+        return render_template("auth/reset-password.html",
+                               email_address=email_address,
+                               form=form,
+                               token=token,
+                               **template_data), 400
+
+
+@main.route('/create-account', methods=["GET"])
```

Comment:
I wonder if this makes more sense as /buyers/create so that it matches /admin/create and /suppliers/create